interface/globals.php

```diff
@@ -130,8 +130,11 @@
         }
     }
 
+    // for both REST API and browser access we can't proceed unless we have a valid site id.
+    // since this is user provided content we need to escape the value but we use htmlspecialchars instead
+    // of text() as our helper functions are loaded in later on in this file.
     if (empty($tmp) || preg_match('/[^A-Za-z0-9\\-.]/', $tmp)) {
-        die("Site ID '". text($tmp) . "' contains invalid characters.");
+        die("Site ID '". htmlspecialchars($tmp, ENT_NOQUOTES) . "' contains invalid characters.");
```

Comment:
I thinking this isn't the right place to do this. Best to qualify the request in the rest interface and return a 401 or 404 back to client. Not sure how we would do that before globals is loaded. Or, one really should not echo back where a login or credential check fails. Just saving hackers time:). So really, that error in globals should be Invalid Url...

Reply:
This change makes sense (no matter what do for the rest interface) since text() is being called before it is even defined. I'd bring this change in for core OpenEMR fix and then work on a rest interface solution separately.

Reply:
If I knew enough of the system I would do what you suggested @sjpadgett as that would be my preference from a security and just good API best practice.  For the new developer I'd also throw the error into the error log so the developer could know what went wrong while keeping information from the hacker.  I'm just learning the system however, so I did the minimum to avoid the fatal error and a more comprehensive overhaul of the REST process flow is warranted.

Reply:
@sjpadgett Reading your request I see you want me to just change the verbiage to invalid url.  I can make that change and still log the error for the developer.

Reply:
Nope just leave as is and we'll let the security folks determine after review all our similar messages.
If you're done, i'll bring in..

Reply:
I guess I was too aggressive :)  I went ahead and changed it to be the invalid url format and pushed it while logging the error.  I won't touch it at this point and y'all can tell me how to proceed.

Reply:
It'd be fine. I'll just get my way on this one:) I'll bring in after you fix travis.
FOUND 4 ERRORS AFFECTING 2 LINES
----------------------------------------------------------------------------------
 137 | ERROR | [x] Spaces must be used to indent lines; tabs are not allowed
 137 | ERROR | [x] Line indented incorrectly; expected at least 8 spaces, found 4
 139 | ERROR | [x] Spaces must be used to indent lines; tabs are not allowed
 139 | ERROR | [x] Line indented incorrectly; expected at least 8 spaces, found 4
----------------------------------------------------------------------------------
PHPCBF CAN FIX THE 4 MARKED SNIFF VIOLATIONS AUTOMATICALLY

Reply:
That's what I get for using VIM to do a quick fix.